### DISCUSSION TITLE
Not downloading pdf / Livewire

### DISCUSSION THREAD

**Discussion by seabasss** [spatie/browsershot#673] [General]
This looks like a great package! Are there any additional steps to make this work with Livewire? I'm trying this, but no file is downloading...

```php
public function exportPDF() {
    return Browsershot::html('<strong>test</strong>')->savePdf('example.pdf');
}
```

**Comment by abbasmashaddy72** [comment]
This will only save the file, you have to do something like this to view that PDF.

`
public function exportPDF() {
    Browsershot::html('<strong>test</strong>')->savePdf('example.pdf');
    return redirect(url('/') . '/example.pdf');
}
`

**Reply by seabasss** [reply]
Thanks! Is that saving it on the server? I don’t need that, I just want it in the users download folder on their computer. Thank you!

**Reply by abbasmashaddy72** [reply]
Then you should directly serve it by pdf() at last without using save option for more on this please read docs

**Reply by seabasss** [reply]
Thanks! Any intel on best approach to accomplish this with laravel livewire? I got it working with streamDownload, but not sure it’s the right approach and it’s kind of slow to export, but maybe that’s just the nature of generating pdf.

**Reply by abbasmashaddy72** [reply]
Thanks, I did not know about streamDownload approach.
